esper/controllers/secureadb/secureadb.py

```diff
@@ -0,0 +1,458 @@
+                return response.json().get("device_certificate")
+
+            time.sleep(1)
+
+        raise SecureADBWorkflowError(f"Failed to acquire Device's certificate in {timeout} secs")
+
+    def setup_ssl_connection(self,
+                             host: str,
+                             port: int,
+                             client_cert: str,
+                             client_key: str,
+                             device_cert: str) -> ssl.SSLSocket:
+        """
+        Create a SSL connection to given host/port endpoint using Mutual TLS, ie,
+        by using device (public-key) certificate as CA certs and Client Certificate
+        and private key as client-side certificate
+
+        :param host: IP or FQDN string of the TCP relay's endpoint
+        :param port: Port on which the TCP relay is listening
+        :param client_cert: File path to Client's Public Key (PEM Formatted)
+        :param client_key: File path to Client's Private Key (PEM Formatted)
+        :param device_cert: File path to Device's Public Key (PEM Formatted)
+        :return: A Secure TCP Socket, wrapped in SSL Context
+        """
+
+        self.app.log.debug("[remoteadb-connect] Starting SSL Connection setup")
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect((host, port))
+
+        self.app.log.debug(f"[remoteadb-connect] Connected to TCP endpoint")
+
+        self.app.log.debug("[remoteadb-connect] Setting up SSL context")
+        context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+        context.verify_mode = ssl.CERT_REQUIRED
+        context.check_hostname = False
+
+        self.app.log.debug("[remoteadb-connect] Loading Device Certificate")
+        context.load_verify_locations(cafile=device_cert)
+        self.app.log.debug("[remoteadb-connect] Loading Client Certificates")
+        context.load_cert_chain(certfile=client_cert, keyfile=client_key)
+
+        self.app.log.debug("[remoteadb-connect] Wrapping SSL Context")
+        secure_sock = context.wrap_socket(sock, server_side=False)
+
+        cert = secure_sock.getpeercert()
+        self.app.log.debug(f"[remoteadb-connect] Peer Certificate -> {cert}")
+
+        # verify server
+        # if not cert or ('commonName', 'test') not in cert['subject'][3]:
+        #     raise Exception("ERROR")
+
+        return secure_sock
+
+    @ex(help='Setup and connect securely via Remote ADB to device',
+        arguments=[
+            (['-d', '--device-name'],
+             {'help': "Device name",
+              'action': 'store',
+              'dest': 'device_name',
+              'default': None})
+        ])
+    def connect(self):
+        """Setup and connect securely via Remote ADB to device"""
+
+        validate_creds_exists(self.app)
+        db = DBWrapper(self.app.creds)
+
+        # client = APIClient(db.get_configure()).get_device_api_client()
+        enterprise_id = db.get_enterprise_id()
```

Comment:
This piece of code from 251-255 seems to be repeated in `_fetch_device_by_name`. Remove if not required at both places.

Reply:
removed the extra line.